src/convex/ConvexAvatarMultiToken.sol

```diff
@@ -579,6 +596,38 @@ contract ConvexAvatarMultiToken is BaseAvatar, ConvexAvatarUtils, PausableUpgrad
         }
     }
 
+    /// @notice Withdraws curve lp token into underlyings proportionally
+    /// @param _lpToken Curve lp address
+    /// @param _lpAmount Amount of bpt to be withdrawn
+    function _withdrawLpToUnderlyings(address _lpToken, uint256 _lpAmount) internal {
+        ICurvePool pool = ICurvePool(META_REGISTRY.get_pool_from_lp_token(_lpToken));
+        uint256 coins = META_REGISTRY.get_n_coins(address(pool));
+
+        if (coins == TWO_COINS_POOL) {
+            uint256[2] memory minOutAmounts;
+            // @audit DAO is accepting under emergency situations for speed defaulting
+            // the amount out expected to zero in this ocassion
+            pool.remove_liquidity(_lpAmount, minOutAmounts);
```

Comment:
can be moved to L618 and reduce redundancy of L612, 613, 615, 616

Reply:
and what about declaring `uint256[coins] memory minOutAmounts;`?

Reply:
in solidity the array length is expecting either an integer literal or constant, cannot use in this case *coins*. thats why there is the redundancy of the method in each if/else block, even tho the signature are not the same